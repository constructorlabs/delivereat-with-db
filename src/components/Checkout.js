import React from 'react';
import { CSSTransitionGroup } from 'react-transition-group';
import '../styles/Checkout.scss';

function Checkout({orderId, changeStage}) {

  return (
    <div className='checkout' onClick={()=>changeStage('menu')}>
      <p className='checkout__logo'>Zing</p>
      <p className='checkout__text'>
        Thank you for placing an order with Zing. Your order
        number is {orderId}. One of our surprisingly attractive 
        baristas is preparing your coffee now.
      </p>
    </div>
  );
}

export default Checkout;