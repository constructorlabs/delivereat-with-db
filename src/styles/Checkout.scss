@import 'defaults';

.checkout {
  height: 100vh;
  background-image: url('/static/assets/checkout-small.jpg');
  background-position: center;
  background-repeat: no-repeat; 
  background-size: cover;

  &__logo {
    position: absolute;
    bottom: 6rem;
    left: 1.6rem;
    color: white;
    font-size: 3rem;
    margin: 0;
    font-family: 'Sedgwick Ave Display', cursive;
  }

  &__text {
    font-family: 'Roboto';
    font-size: 1rem;
    position: absolute;
    bottom: 1.5rem;
    left: 1.6rem;
    color: white;
    width: 80%;
  }
}

.checkout-appear {
  opacity: 0.01;
}

.checkout-appear.checkout-appear-active {
  opacity: 1;
  transition: opacity 1s ease-in;
}

.checkout-leave {
  opacity: 1;
}

.checkout-leave.checkout-leave-active {
  opacity: 0.01;
  transition: opacity 1s ease-in;
}

@media screen and (min-width: 768px) {

  .checkout {
    background-image: url('/static/assets/checkout-medium.jpg');

    &__logo {
      font-size: 5rem;
      bottom: 10rem;
      left: 3rem;
    }

    &__text {
      font-size: 1.4rem;
      bottom: 5rem;
      left: 3rem;
    }
  }
}

@media screen and (min-width: 960px) {

}
