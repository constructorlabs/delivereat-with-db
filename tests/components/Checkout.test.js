import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Checkout from '../../src/components/Checkout';

const orderId = 10;
const mockChangeStage = jest.fn();

describe('Checkout', () => {
  test('matches the snapshot', () => {
    const tree = renderer
      .create(<Checkout orderId={orderId} changeStage={mockChangeStage} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('click changes stage', () => {
    const wrapper = shallow(<Checkout orderId={orderId} changeStage={mockChangeStage} />);
    wrapper.find('.checkout').simulate('click');
    expect(mockChangeStage.mock.calls).toEqual([['menu']]);
  });
});
