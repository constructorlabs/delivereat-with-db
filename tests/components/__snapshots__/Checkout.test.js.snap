// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout matches the snapshot 1`] = `
<div
  className="checkout"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <p
    className="checkout__logo"
  >
    Zing
  </p>
  <p
    className="checkout__text"
  >
    Thank you for placing an order with Zing. Your order number is 
    10
    . One of our surprisingly attractive baristas is preparing your coffee now.
  </p>
</div>
`;
